.github/workflows/build-test-images.yml

```
@@ -8,17 +8,56 @@ on:
   workflow_dispatch:
 
 jobs:
-  build_dev_images:
+  prepare_pgversion_matrix:
+    runs-on: ubuntu-latest
+    steps:
+      # Need the repo checked out in order to read the file
+      - uses: actions/checkout@v3
+      - id: get_versions
+        run: cat circleci/images/PG_VERSIONS >> $GITHUB_OUTPUT
+    outputs:
+      # Will look like '["13.9", "14.6", "15.1"]'
+      pgversions: ${{ toJSON(steps.get_versions.outputs.*) }}
+
+  build_dev_images_per_postgres:
     if: github.ref_name != 'master'
     name: push-test-images-for-dev
     runs-on: ubuntu-latest
+    needs:
+      - prepare_pgversion_matrix
     strategy:
       fail-fast: false
       matrix:
         command:
           - extbuilder
           - exttester
           - failtester
+        pgversion: ${{ fromJSON(needs.prepare_pgversion_matrix.outputs.pgversions) }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
```

I can see that v2 is out. Any particular reason we want to use the older one?

```suggestion
        uses: docker/login-action@v2
```

It may be a nice idea to actually setup dependabot to keep the actions up-to-date. A sample `.github/dependabot.yml` file is as follows:

```yml
version: 2
updates:
  # Maintain dependencies for GitHub Actions
  - package-ecosystem: "github-actions"
    directory: "/"
    schedule:
      interval: "daily"
```

Although I am fairly sure that dependabot would keep silent unless there is a known issue with `docker/login-action@v1` or similar outdated action versions.